asmcomp/linearize.ml

```diff
@@ -218,48 +223,48 @@ let linear i n contains_calls =
             | _ -> Cmm.new_label ())
             handlers in
         let exit_label_add = List.map2
-            (fun (nfail, _) lbl -> (nfail, (lbl, !try_depth)))
+            (fun (nfail, _) lbl -> (nfail, (lbl, try_depth)))
             handlers labels_at_entry_to_handlers in
-        let previous_exit_label = !exit_label in
-        exit_label := exit_label_add @ !exit_label;
+        let previous_exit_label = List.append exit_label exit_label_add in
```

Comment:
Just reuse the same names as before your patch:
```ocaml
let exit_label = exit_label_add @ exit_label in
```

This will make the diff easier to read and will hopefully fix the bugs that the CI complains about.
If you look at the rest of the code, `previous_exit_label` was only used to restore the contents of the reference, so because of the switch to parameters we don't need to remember it at all.